widget/view.controller.js

```diff
@@ -2,38 +2,186 @@
+        "keystore": "sfsp-excludelist-ports",
+        "defaultValue": ""
+      },
+      "CIDR_Range": {
+        "keystore": "sfsp-cidr-range",
+        "defaultValue": "10.0.0.0/8,172.16.0.0/12,192.168.0.0/16"
+      }
+    };
+    $scope.defaultGlobalSettings = {};
+    $scope.initList = [];
+    $scope.updatedGlobalSettings = {};
+    $scope.moveNext = moveNext;
+    $scope.moveBack = moveBack;
+    $scope.isLightTheme = $rootScope.theme.id === 'light';
+    $scope.startPageImage = $scope.isLightTheme ? widgetBasePath + 'images/sfsp-start-light.png' : widgetBasePath + 'images/sfsp-start-dark.png';
+    $scope._buildPayload = _buildPayload;
+    $scope.modifyGlobalSettings = modifyGlobalSettings;
+    $scope.commitGlobalSettings = commitGlobalSettings;
+    $scope.test = test;
+
+
+    function _handleTranslations() {
+      widgetUtilityService.checkTranslationMode($scope.$parent.model.type).then(function () {
+        $scope.viewWidgetVars = {
+          // Create your translating static string variables here
+        };
+      });
+    }
+
+
+    function _buildPayload(keyName, keyValue, action) {
+      if (action === 'createKeyStore') {
+        var apiPayload = soarConfigService.constants().createKeyStorePayload;
+        apiPayload['key'] = keyName;
+        apiPayload['jSONValue'] = keyValue[0].length > 0 ? keyValue : '';
       }
+      if (action === 'findKeyStore') {
+        var apiPayload = soarConfigService.constants().findKeyStorePayload;
+        apiPayload['filters'][0]['value'] = keyName;
+      }
+      return apiPayload;
+    }
 
-      function moveNext() {
 
-        WizardHandler.wizard('soarFrameworkConfigurationWizard').next();
+    function _handleGblVarsAndKeyStores() {
+      Object.keys($scope.gblVarToKeyStoreMapping).forEach(function (item) {
+        if (item === 'CIDR_Range') {
+          var payload = _buildPayload('sfsp-cidr-range', null, 'findKeyStore');
+          soarConfigService.getKeyStoreRecord(payload, 'keys').then(function (response) {
+            var resp = response['hydra:member'];
+            if (resp.length === 0) {
+              var keyValue = $scope.gblVarToKeyStoreMapping['CIDR_Range'].defaultValue.split(',');
+              var payload = _buildPayload('sfsp-cidr-range', keyValue, 'createKeyStore');
+              soarConfigService.createOrUpdateKeyStore(payload, 'keys').then(function (res) {
+                $scope.defaultGlobalSettings[res.key] = { 'recordValue': res.jSONValue, 'recordUUID': res.uuid };
+              });
+            }
+            else {
+              $scope.defaultGlobalSettings[resp[0].key] = { 'recordValue': resp[0].jSONValue, 'recordUUID': resp[0].uuid };
+            }
+          });
+        }
+        else {
+          soarConfigService.getGBLVariable(item).then(function (response) {
+            // Check if exclude list global variable already present 
+            var resp = response['hydra:member'];
+            if (resp.length > 0) {
```

Comment:
use this condition (response['hydra:member'] && response['hydra:member'].length > 0)

Reply:
DONE